### ISSUE TITLE
MdcTheme crashes when running a preview

### ISSUE THREAD

**Issue by manuelvicnt** [material-components/material-components-android-compose-theme-adapter#31]
Running a preview on emulator that uses `MdcTheme` crashes with the following error:

```
Caused by: java.lang.IllegalArgumentException: MaterialThemeUsingMdcTheme requires the host context's theme to extend Theme.MaterialComponents
 at com.google.android.material.composethemeadapter.MdcTheme.createMdcTheme(MdcTheme.kt:134)
 at com.google.android.material.composethemeadapter.MdcTheme.createMdcTheme$default(MdcTheme.kt:125)
 at com.google.android.material.composethemeadapter.MdcTheme.MdcTheme(MdcTheme.kt:75)
```

#### How to reproduce

Open the `MigrationCodelab` project from the [android-compose-codelabs `end` branch](https://github.com/googlecodelabs/android-compose-codelabs/tree/end) and run any of the previews in the `PlantDetailDescription.kt` file.

#### Versions
- Compose `alpha06`
- Android Studio 4.2 Canary 13

**Comment by chrisbanes** [comment]
The error is correct, the deployed preview isn't using the correct theme for whatever reason. I don't think this is a bug in the library, but rather the deploy preview feature. Let's review with the tools team and see what they say.

**Comment by svenjacobs** [comment]
I'm still facing this issue with Compose 1.0.0-beta08 and Android Studio Arctic Fox 2020.3.1 Beta 3. How can I use `MdcTheme` in conjunction with preview?

**Comment by ricknout** [comment]
In the above PR I've added a preview to the sample, and it's using Compose 1.0.0-beta08 / AGP 7.0.0-beta03. It's working for me in Android Studio Arctic Fox 2020.3.1 Beta 3, but I did need to refresh the preview layout a few times (so I assume it's a bug in preview as opposed to the library).

**Comment by ChrisTitos** [comment]
When I try to open the sample project of this repo I get the same error. (I don't get it in my personal project).
When I added a new file with a Preview in it, it previewed correctly once, but after editing/refreshing I keep getting this error again.

Android Studio Arctic Fox | 2020.3.1 Patch 3

This was the preview I tried to add in a new file:
```
@Preview
@Composable
fun TypoPreview() {
    Column {
        Text(text = "Hello world", style = MaterialTheme.typography.h3)
        MdcTheme {
            Text(text = "Hello world", style = MaterialTheme.typography.h3)
        }
    }
}
```
(I was trying to see if the fonts of the theme should work, because that's not working in preview in my project)
